addon/mixins/attachable.js

```diff
@@ -75,7 +75,7 @@ export default Ember.Mixin.create({
     }
   },
   _rootKey: function() {
-    return Ember.String.underscore(Ember.String.decamelize(this.constructor.typeKey));
+    return Ember.String.underscore(this._modelName);
```

Comment:
In between the first and second request I noticed [in the release notes](http://emberjs.com/blog/2015/05/21/ember-data-1-0-beta-18-released.html) that:

> typeKey was previously always normalized to be a camelCased string.
> 
> In Ember Data 1.0.0-beta.18, this property is now called modelName. In addition, the 
> modelName is a dasherized string.

Which is why I removed the decamalize call.